### PR TITLE
feat(web): validate password

### LINKED ISSUE #778: 账号密码必填？
### 功能描述

因互联网存在各种扫描，内网也会存在，考虑帐号密码必填，同时验证弱密码

### 解决的问题

_No response_

### 附加信息

_No response_

### 检查清单

- [X] 我已搜索同类问题，并确保没有我要提交的功能
- [X] 我已使用最新版本，并确保该功能仍未在最新版本中实现

### PR DESCRIPTION
**Pull request by WaterLemons2k**
# What does this PR do?

Use [go-password-validator](https://github.com/wagslane/go-password-validator) to checks that the password meets the minimum password requirements.

Fixes #778.

# Motivation

#778

# Additional Notes

https://github.com/wagslane/go-password-validator
https://github.com/coder/coder/blob/0c73164f15da870f0c71d842844a95e4fb73304e/coderd/userpassword/userpassword.go#L128

### CODE REVIEW

**Comment by WaterLemons2k** [comment]
另外，目前如果密码不够强返回的错误说明是英文的，要不要重写 [`Validate()`](https://github.com/wagslane/go-password-validator/blob/v0.3.0/validate.go#L13) 以返回中文错误说明？

**Comment by jeessy2** [comment]
可以返回中文。`minEntropy` 60得不得高了点，选个适中的

File: web/save.go [hunk 1]
```diff
@@ -50,6 +50,12 @@ func checkAndSave(request *http.Request) string {
 
 	}
 
+	// 检查密码是否够强
+	err = validate(passwordNew)
```
Comment (jeessy2) [inline web/save.go:54]: 成了密码必须校验了，输入为空也要校验

Comment (WaterLemons2k) [reply web/save.go]: 已修复如此低级的错误。

**Review by jeessy2 — APPROVED** [review]
thanks @WaterLemons2k

**Pull request merged**